rholang/src/test/scala/coop/rchain/rholang/interpreter/SynchronousOutputSpec.scala

```diff
@@ -0,0 +1,52 @@
+package coop.rchain.rholang.interpreter
+
+import coop.rchain.rholang.interpreter.ParBuilderUtil.assertCompiledEqual
+import org.scalatest.{FlatSpec, Matchers}
+
+class SynchronousOutputSpec extends FlatSpec with Matchers {
+
+  "';'" should "have a higher precedence than '|'" in {
+    val s = "@1!?(1) ; @2!?(2). | @3!?(3)."
+    val t =
+      """
+          # new x0 in {
+          #   @1!(*x0, 1) |
```

Comment:
I am trying `PSendSynch`. I was thinking that `PSendSynch` is supposed to solve condition like below with `ack` thing.
  The example here  https://github.com/rchain/rchain/blob/dev/rholang/examples/stdoutAck.rho#L1-L6
  ```
  new ack, stdout(`rho:io:stdout`), stdoutAck(`rho:io:stdoutAck`) in {
    stdoutAck!("hello, world!", *ack) |
    for (_ <- ack) {
      stdout!("received")
    }
  }
  ```

  can be re-written into
 ```
  new stdout(`rho:io:stdout`), stdoutAck(`rho:io:stdoutAck`) in {
    stdoutAck!?("hello, world!"); stdout!("received")
  }
```
But it seems that `PSendSynch` is quite limited which here the `new x0` must be the first argument in `@1!(*x0, 1)`.
Is that correct?

Reply:
Yes. Unfortunately, in order for remainder syntax to still make sense, we have to switch to the convention that the return channel is the first argument received. So, to be consistent, we'll have to rewrite existing Rholang source to follow that convention. That includes having rho:io:stdoutAck (and the other system processes) expect the return channel to be the first argument. Then your example will work.